flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/input/operator/StateReaderOperator.java

```diff
@@ -104,7 +104,21 @@ public void open() throws Exception {
 	}
 
 	public void close() throws Exception {
-		FunctionUtils.closeFunction(function);
+		Exception exception = null;
+
+		try {
+			FunctionUtils.closeFunction(function);
+		} catch (Exception e) {
+			// The state backend must always be closed
+			// to release native resources.
+			exception = e;
+		}
+
+		keyedStateBackend.dispose();
```

Comment:
how about just:

```
try {
    FunctionUtils.closeFunction(function);
} finally {
    keyedStateBackend.dispose();
}
```

?

Reply:
-1 `keyedStateBackend.dispose()` throws and exception and checkstyle did not like calling a method that throws inside of a `finally` block.